keel-core/src/main/kotlin/com/netflix/spinnaker/keel/resources/ResourceSpecIdentifier.kt

```diff
@@ -12,14 +14,30 @@ import org.springframework.stereotype.Component
  * things like reading resources from the database, or parsing JSON.
  */
 @Component
-class ResourceSpecIdentifier @Autowired constructor(
-  private val kinds: List<SupportedKind<*>>
+class ResourceSpecIdentifier @Autowired constructor (
+  private val extensionRegistry: ExtensionRegistry?
 ) {
+  private var _kinds: List<SupportedKind<*>>? = null
+
+  private val kinds: List<SupportedKind<*>>
+    get() {
+      return when {
+        _kinds != null -> _kinds!!
+        extensionRegistry != null -> extensionRegistry
+          .extensionsOf<ResourceSpec>()
+          .entries
+          .map { SupportedKind(ResourceKind.parseKind(it.key), it.value) }
+        else -> error("Keel is misconfigured. No resource kinds registered.")
```

Comment:
@robfletcher this was the best I could come up with, but it kinda sucks. I didn't want to rewrite all the tests that used the `vararg` constructor... Let me know if you have a better idea.

Reply:
I'd go with keeping the original constructor but moving the `@Autowired` annotation to a new one that takes `ExtensionRegistry` as a parameter.

Reply:
I thought of that, but wasn't sure that the `@Component` annotation on the class/default constructor wouldn't give it priority when auto-wiring...

Reply:
Easy enough to test

Reply:
It works. Sorry for being lazy. 😄